src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/tests/latest/test_iot_pnp_commands.py

```diff
@@ -0,0 +1,102 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=too-many-statements
+
+from azure.cli.testsdk import ScenarioTest
+
+
+class IoTPnPTest(ScenarioTest):
+
+    def test_iot_pnp(self):
+        repo_endpoint = 'https://provider.azureiotrepository-test.com'
```

Comment:
feel free to use [self.kwargs to store constants / variables](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_tests.md#sample-6-random-name-and-name-mapping) to make calls to self.cmd easier.

```suggestion
        self.kwargs.update({
            'endpoint': 'https://provider.azureiotrepository-test.com',
            'name': 'clitestrepo',
            'name_2': 'clitestrepo-updated'
            'role': 'Reader'
            'role_2': 'Admin'
        })
        repo_endpoint = 'https://provider.azureiotrepository-test.com'
```

Reply:
Updated